src/article_id.rs

```diff
@@ -324,6 +324,26 @@ impl Display for ArticleId<'_> {
 	}
 }
 
+impl<'a> ArticleId<'a> {
+	fn try_from_id(id: &'a str) -> Result<Self, ArticleIdError> {
```

Comment:
IMO, it seems like this function would be beneficial if it's public + with a documentation block including examples.

Reply:
Hmm, I don't think `try_from_id_unchecked` would be right, since the `unchecked` suffix convention implies there's no validation done. 

`try_from_raw_id` feels like a slightly odd description but succinct; something more accurate would be `try_from_unprefixed_id` although it'd be a bit longer.

Reply:
How about try_from_short_id?

Reply:
That works. The documentation block should add a note about what makes it different from `try_from(&str)` as well